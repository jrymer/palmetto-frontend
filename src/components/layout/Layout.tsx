import React from 'react';
import styled, { ThemeProvider } from 'styled-components';

import theme from './theme';

const Container = styled.div`
  display: flex;
  height: 100vh;
  width: 100vw;
  background-color: ${(props) => props.theme.colors.background};
  overflow: hidden;
`;

interface LayoutProps {
  children: React.ReactNode;
}

/**
 * Wraps our app in a theme provider and put app wide layout elements (like a non existent footer and navbar) here
 *
 * @param {*} { children }
 * @return {*}
 */
const Layout: React.FC<LayoutProps> = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <Container>{children}</Container>
    </ThemeProvider>
  );
};

export default Layout;
