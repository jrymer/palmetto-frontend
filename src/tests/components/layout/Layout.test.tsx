import React from 'react';

import Layout from '../../../components/layout/Layout';
import { createSnapshot, render, screen } from '../../testWrapper';

const child = <div>child</div>;

describe('Layout tests', () => {
  it('Layout snapshot is valid', () => {
    const snapshot = createSnapshot(<Layout>{child}</Layout>);
    expect(snapshot).toMatchSnapshot();
  });
  it('Renders a child', () => {
    render(<Layout>{child}</Layout>);
    expect(screen.getByText(/child/i)).toBeInTheDocument();
  });
});
