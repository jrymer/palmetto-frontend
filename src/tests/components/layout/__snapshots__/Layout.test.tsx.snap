// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout tests Layout snapshot is valid 1`] = `
<div
  className="sc-bczRLJ ljbnaa"
>
  <div>
    child
  </div>
</div>
`;
